fv3core/stencils/remapping.py

```diff
@@ -1,128 +1,521 @@
+
+        self._pressures_mapu = FrozenStencil(
+            pressures_mapu, origin=grid.compute_origin(), domain=self._domain_jextra
+        )
+
+        self._kord_mt = self.namelist.kord_mt
+        self._map_single_u = MapSingle(
+            self._kord_mt, -1, grid.is_, grid.ie, grid.js, grid.je + 1
+        )
+
+        self._pressures_mapv = FrozenStencil(
+            pressures_mapv,
+            origin=grid.compute_origin(),
+            domain=(grid.nic + 1, grid.njc, grid.npz + 1),
+        )
+
+        self._map_single_v = MapSingle(
+            self._kord_mt, -1, grid.is_, grid.ie + 1, grid.js, grid.je
+        )
+
+        self._update_ua = FrozenStencil(
+            update_ua, origin=grid.compute_origin(), domain=self._domain_jextra
+        )
+
+        self._copy_from_below_stencil = FrozenStencil(
+            copy_from_below,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(),
+        )
+
+        self._moist_cv_last_step_stencil = FrozenStencil(
+            moist_pt_last_step,
+            origin=(self.grid.is_, self.grid.js, 0),
+            domain=(self.grid.nic, self.grid.njc, self.grid.npz + 1),
+        )
+
+        self._basic_adjust_divide_stencil = FrozenStencil(
+            adjust_divide_stencil,
+            origin=self.grid.compute_origin(),
+            domain=self.grid.domain_shape_compute(),
+        )
+
+        self._do_sat_adjust = self.namelist.do_sat_adj
+
+        self.kmp = self.grid.npz - 1
+        for k in range(pfull.shape[0]):
+            if pfull[k] > 10.0e2:
+                self.kmp = k
+                break
+
+        self._saturation_adjustment = SatAdjust3d(self.kmp)
+
+        self._sum_te_stencil = FrozenStencil(
+            sum_te,
+            origin=(self.grid.is_, self.grid.js, self.kmp),
+            domain=(self.grid.nic, self.grid.njc, self.grid.npz - self.kmp),
+        )
+
+    def __call__(
+        self,
+        tracers: Dict[str, "FloatField"],
+        pt: FloatField,
+        delp: FloatField,
+        delz: FloatField,
+        peln: FloatField,
+        u: FloatField,
+        v: FloatField,
+        w: FloatField,
+        ua: FloatField,
+        va: FloatField,
+        cappa: FloatField,
+        q_con: FloatField,
+        q_cld: FloatField,
+        pkz: FloatField,
+        pk: FloatField,
+        pe: FloatField,
+        hs: FloatFieldIJ,
+        te0_2d: FloatFieldIJ,
+        ps: FloatFieldIJ,
+        wsd: FloatFieldIJ,
+        omga: FloatField,
+        ak: FloatFieldK,
+        bk: FloatFieldK,
+        pfull: FloatFieldK,
+        dp1: FloatField,
+        ptop: float,
+        akap: float,
+        zvir: float,
+        last_step: bool,
+        consv_te: float,
+        mdt: float,
+        bdt: float,
+        do_adiabatic_init: bool,
+        nq: int,
+    ):
+        """
+        Remap the deformed Lagrangian surfaces onto the reference, or "Eulerian",
+        coordinate levels.
+        """
```

Comment:
[optional, not sure if we are doing this strictly in this stage of the refactor] Add documentation for each variable. I think you can find most of the descriptions from other classes. The fortran code also has some documentation as well.

Reply:
I listed the variables that were documented from `fv_mapz.F90`.